prov/rxd/src/rxd_cq.c

```diff
@@ -329,7 +329,10 @@ static int rxd_send_cts(struct rxd_ep *rxd_ep, struct rxd_rts_pkt *rts_pkt,
 	struct rxd_cts_pkt *cts;
 	int ret = 0;
 
-	rxd_update_peer(rxd_ep, peer, rts_pkt->rts_addr);
+	if (dlist_empty(&rxd_ep->peers[peer].unacked) ||
+	    rxd_get_base_hdr(container_of((&rxd_ep->peers[peer].unacked)->next,
+			     struct rxd_pkt_entry, d_entry))->type != RXD_RTS)
+		rxd_update_peer(rxd_ep, peer, rts_pkt->rts_addr);
```

Comment:
The commit message for this change suggests a deeper problem.  I will check directly.